### PR TITLE
Added metal packages to extra setup.py and release the metal buffer on del

### PR DESCRIPTION
**Pull request by dc-dc-dc**

### CODE REVIEW

File: tinygrad/runtime/metal.py [hunk 1]
```diff
@@ -22,6 +22,8 @@ def __init__(self, shape): raise NotImplementedError("Metal runtime doesn't supp
 
 class CLBuffer:
   def __init__(self, size): self._cl = device.newBufferWithLength_options_(size, Metal.MTLResourceStorageModeShared)
+  def __del__(self): self._cl.release()
```
Comment (geohot) [inline tinygrad/runtime/metal.py:25]: This doesn't happen by default on out of scope?

Comment (dc-dc-dc) [reply tinygrad/runtime/metal.py:25]: didnt dive deep into it, but noticed that test_speed_v_torch execution time reduces after adding that
without release ~24 seconds
with release ~19 seconds

Comment (geohot) [reply tinygrad/runtime/metal.py:25]: Repeatability? I can't see why this would have an affect.

Comment (dc-dc-dc) [reply tinygrad/runtime/metal.py:25]: yes, just ran some tests now and watched the memory usage and without releasing it kept growing

Comment (geohot) [reply tinygrad/runtime/metal.py:25]: Which raises deeper questions about pyobic objects...will merge

**Pull request merged**